### ISSUE TITLE
epoch is ignored when checking whether to upgrade a package

### ISSUE THREAD

**Issue by biinari** [aurapm/aura#402]
[`epoch`](https://wiki.archlinux.org/index.php/PKGBUILD#epoch) is used to force packages to be seen as newer than versions with a lower epoch. As aura does not take any heed of epoch, it fails to notice when these packages need upgrading.

The package I noticed this with is `php-pear` which is currently at version 5.6.16-3 (epoch 0 by default). I ran the upgrade directly with makepkg. The new version is 1:1.10.1-1 (epoch 1, pkgver 1.10.1, pkgrel 1).

The info reported by `pacman -Qi` may be useful.

``` sh
# before php-pear upgrade
$ pacman -Qi php-pear | grep '^Version'
Version         : 5.6.16-3

# after php-pear upgrade
$ pacman -Qi php-pear | grep '^Version'
Version         : 1:1.10.1-1
```

If this is more effort than it is worth, then I don't mind leaving it. As per the warning in the PKGBUILD wiki, epochs are only supposed to be used in exceptional circumstances.

**Comment by fosskers** [comment]
Is `epoch` an Arch Linux packaging convention, or a php one? In all cases we recommend [Semantic Versioning](semver.org), where a non-backward-compatible release can be signalled by incrementing the major (first) version number. 

`5.6.16 -> 6.0.0`

**Comment by fosskers** [comment]
It seems that `epoch` can be specified in a PKGBUILD directly as you mentioned. Aura 1.4 will take care of this.

**Comment by biinari** [comment]
Thanks @fosskers. Yes it's a PKGBUILD variable used when the versioning scheme of the upstream package has been changed incompatibly. One other example I have seen somewhere (if only I could remember where) was switching between semantic versioning and date/time stamps.

I haven't got a copy of the old PKGBUILD, but it looks like php-pear used to use the PHP version instead of the PEAR version, hence the version of the aur package is now lower and the need for the epoch bump.

**Comment by fosskers** [comment]
If you need the upgrade to work right-this-moment, you can always reinstall the package manually with `-A`. Then on the next upgrade it should work properly.

**Comment by fosskers** [comment]
Epoch numbers are no longer ignored on `master`, which will be released soon as Aura 2.0.
